portal-impl/src/com/liferay/portlet/asset/service/impl/AssetVocabularyLocalServiceImpl.java

```diff
@@ -389,6 +401,49 @@ public AssetVocabulary getGroupVocabulary(long groupId, String name)
 		return assetVocabularyPersistence.findByG_N(groupId, name);
 	}
 
+	@Override
+	public AssetVocabulary getOrAddIncompleteVocabulary(
+			String externalReferenceCode, long userId, long groupId)
+		throws PortalException {
+
+		AssetVocabulary assetVocabulary =
+			fetchAssetVocabularyByExternalReferenceCode(
+				externalReferenceCode, groupId);
+
+		if (assetVocabulary != null) {
+			return assetVocabulary;
+		}
+
+		if (!LazyReferencingThreadLocal.isEnabled()) {
+			throw new NoSuchVocabularyException(
+				StringBundler.concat(
+					"Unable to find asset vocabulary with external reference ",
+					"code ", externalReferenceCode, " and group ", groupId));
+		}
+
+		assetVocabulary = assetVocabularyLocalService.fetchGroupVocabulary(
+			groupId, externalReferenceCode);
+
+		if (assetVocabulary != null) {
+			throw new DuplicateVocabularyException(
+				"A category vocabulary with the name " + externalReferenceCode +
```

Comment:
also here the two error messages have different naming convention

Reply:
I copied the same message that is on the Class

https://github.com/liferay/liferay-portal/blob/e24afed1f539990edaa41ea4f43118f95c77e607/portal-impl/src/com/liferay/portlet/asset/service/impl/AssetVocabularyLocalServiceImpl.java#L633

Reply:
if it's your standard probably not, they should be fine